Library/Formula/aws-sdk-cpp.rb

```diff
@@ -0,0 +1,70 @@
+class AwsSdkCpp < Formula
+  desc "AWS SDK for C++"
+  homepage "https://github.com/aws/aws-sdk-cpp"
+  url "https://github.com/aws/aws-sdk-cpp/archive/0.9.6.tar.gz"
+  sha256 "cf2dad62e50c5c5cff88a4ad959af3aa402aa7e8aa1692bdf807052c0fe4aeb9"
+  head "https://github.com/aws/aws-sdk-cpp.git"
+
+  ALL_CLIENTS = [
```

Comment:
Homebrew doesn't allow that sort of options handling, unfortunately.  In the formula you specify an option one of two ways:
```ruby
option "with-foo", "Build with foo"
  # build.with?("foo") => false unless `brew install --with-foo`

option "without-foo", "Build without foo" 
  # build.with?("foo") => true unless `brew install --without-foo`
```
So if all the options are on by default, that means they'll all be specified in the `option "without-#{client}"` form, and users who want the minimal build will have to pass `--without-#{client}` for each.